soh/src/code/z_message_PAL.c

```diff
@@ -1218,6 +1227,20 @@ void Message_LoadItemIcon(PlayState* play, u16 itemId, s16 y) {
     MessageContext* msgCtx = &play->msgCtx;
     InterfaceContext* interfaceCtx = &play->interfaceCtx;
 
+    // itemId gets formatted as a char in CustomMessage::ITEM_OBTAINED
+    // ITEM_INVALID_4 happens to turn into "~" which is cool so let's use that 
+    if (itemId == ITEM_INVALID_4) {
+        R_TEXTBOX_ICON_XPOS = R_TEXT_INIT_XPOS - sIconItem32XOffsets[gSaveContext.language];
+        R_TEXTBOX_ICON_YPOS = y + 6;
+        R_TEXTBOX_ICON_SIZE = 32;
+        uint8_t customIconIndex = msgCtx->msgBufDecoded[msgCtx->msgBufPos + 2];
+        memcpy((uintptr_t)msgCtx->textboxSegment + MESSAGE_STATIC_TEX_SIZE,
+            gCustomGetItemIcons[customIconIndex], strlen(gCustomGetItemIcons[customIconIndex]) + 1);
+        msgCtx->msgBufPos += 2;
+        msgCtx->choiceNum = 1;
+        return;
+    }
```

Comment:
basically just doing everything from the `itemId < ITEM_MEDALLION_FOREST` block but incrementing the msgbuf by 2 because we're using that char for the custom item index

also `ITEM_INVALID_4 > ITEM_MEDALLION_FOREST` so the return keeps us out of that block